File: src/main/scala/com/stone/spark/operation/FilterOperator.scala
package com.stone.spark.operation

//import org.apache.spark.sql.SparkSession
import org.apache.spark.{SparkConf, SparkContext}

object FilterOperator {

  def main(args: Array[String]): Unit = {
    var sparkConf = new SparkConf().setAppName("FilterOperator").setMaster("local")
    var sc = new SparkContext(sparkConf)


    val numbers = List(1, 2, 3, 4, 5)
    var numRdd = sc.parallelize(numbers)
        .filter(f => f%2 ==0)


    numRdd.foreach(f => println("偶数为: " + f))
  }
}

File: src/main/java/com/stone/spark/operator/FilterOperator.java
package com.stone.spark.operator;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.api.java.function.VoidFunction;

import java.util.Arrays;
import java.util.List;

public class FilterOperator {

    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setAppName("FilterOperator").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);

        List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5);

        JavaRDD<Integer> filterRDD =  sc.parallelize(numbers)
                .filter(new Function<Integer, Boolean>() {
                    public Boolean call(Integer v1) throws Exception {
                        return v1 % 2==0;
                    }
                });

        filterRDD.foreach(new VoidFunction<Integer>() {
//            @Override
            public void call(Integer integer) throws Exception {
                System.out.println("整数: " + integer);
            }
        });
    }
}
